### PR TITLE
Added Implementation of CRT in JavaScript

### LINKED ISSUE #1878: Implementation of CRT (Chinese Remainder Theorem in JS)
@asha15 can I take this up , there is no implementation in JavaScript.

### PR DESCRIPTION
**Pull request by abhu-A-J**
Fixes #1878 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- The Implementation of CRT commonly known as Chinese Remainder Theorem in JavaScript.

#### Languages Used:
- JavaScript
 

#### Files Added:
- Addition of file named Chinese_Remainder_Theorem.js inside the Chinese_Remainder_Theorem folder.

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by abhu-A-J** [comment]
@asha15 Can you review this.

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+3. remianders(i) is the remiander set value.
+4. divisors(i) is the divisor set value.
+5. M is known as the dynamic range
+*/
+
+//the inverse element is defined by (s1*s1_inverse)%moduli =1, where s1_inverse needs to be evaluated.
+const inverseElement = (s1, moduli) => {
+    let x_primary = 0;
+    let x_secondary = 1;
+    let moduli_primary = moduli;
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: keep a blank line after declaring variable

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if (moduli === 1) return 0;
+    while (s1 > 1) {
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: keep a blank line after the if statement

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        let quotient = Math.floor(s1 / moduli);
+        let temp = moduli;
+        moduli = s1 % moduli;
+        s1 = temp;
+        temp = x_primary;
+        x_primary = x_secondary - quotient * x_primary;
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: It's better if you could give a commend regarding the operation

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        x_secondary = temp;
+    }
+    if (x_secondary < 0) {
+        x_secondary += moduli_primary;
+    }
+    return x_secondary;
+};
+
+const CRT = (divisors, remainders) => { //take the divisors and remainder set as arrays
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Take this commend into the previous line

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    const lengthDivisors = divisors.length;
+    const lengthRemainders = remainders.length;
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: keep a line space

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    if (lengthRemainders !== lengthDivisors) {
+        //if they dont match return since result cannot be evaluated
+        return `The length of remainder set and moduli set don't match,result cannot be evaluated`;
+    }
+    let M = 1; //M is the dynamic range evaluated as the product of all divisors.
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: move this commend into previous line n also keep a blank line after if statement

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    for (let i = 0; i < lengthDivisors; i++) {
+        M = M * divisors[i];
+    }
+    let minDividend = 0; //the minium divident that is evalued using CRT
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: change the commend into previous line

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    for (let i = 0; i < lengthDivisors; i++) {
+        let s1 = Math.floor(M / divisors[i]); //partial product is defined as M/divisors[i], and represented by s1
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: here too

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 1 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: I replaced this mentioned with a more simpler method. I hope that's okay now

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 2 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Done with this change

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 4 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Done with this change

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 5 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Done with this change

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 6 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Done with this change

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 7 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Done with this change

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js (hunk 8 above)
Comment (abhu-A-J) [reply Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: Done with this change

**Comment by abhu-A-J** [comment]
@asha15 I have done required changes as requested. You can review it again in your free time

File: Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    //go from 1 upto moduli, and check the condition for inverse mentioned above
+    for (let counter = 1; counter < moduli; counter++) {
+        if ((s1 * counter) % moduli === 1) {
+            return counter;
+        }
+    }
+}
+
+//take the divisors and remainder set as arrays
+
```
Comment (asha15) [inline Chinese_Remainder_Theorem/Chinese_Remainder_Theorem.js]: remove this line. rest is looking good

**Comment by abhu-A-J** [comment]
@asha15 Ma'am I have removed the line you asked. Everything is looking fine now??

**Comment by asha15** [comment]
@abhu-A-J  yes, will go for merge

**Comment by abhu-A-J** [comment]
@asha15 ma'am thanks for merge

**Pull request merged**